crates/syntax/src/syntax_editor/edits.rs

```diff
@@ -0,0 +1,72 @@
+//! Structural editing for ast using `SyntaxEditor`
+
+use crate::{
+    ast::make, ast::AstNode, ast::Fn, ast::GenericParam, ast::HasGenericParams, ast::HasName,
+    syntax_editor::Position, syntax_editor::SyntaxEditor, SyntaxKind,
+};
+
+impl SyntaxEditor {
+    /// Adds a new generic param to the function using `SyntaxEditor`
+    pub fn syntax_editor_add_generic_param(&mut self, function: Fn, new_param: GenericParam) {
```

Comment:
last nit :) 
```suggestion
    pub fn add_generic_param(&mut self, function: Fn, new_param: GenericParam) {
```

you can probably also take `&Fn`

Reply:
> you can probably also take &Fn

Yes